ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java

```diff
@@ -650,19 +723,40 @@ private String generateRandomString(int length) {
    * Any other headers have to be set explicitly.
    * @param headers the {@link JSONObject} where the headers should be set.
    * @param ttlInSecs sets the {@link RestUtils.Headers#TTL} header.
-   * @param isPrivate sets the {@link RestUtils.Headers#PRIVATE} header. Allowed values: true, false.
    * @param serviceId sets the {@link RestUtils.Headers#SERVICE_ID} header.
    * @param contentType sets the {@link RestUtils.Headers#AMBRY_CONTENT_TYPE} header.
    * @param ownerId sets the {@link RestUtils.Headers#OWNER_ID} header. Optional - if not required, send null.
+   * @param insertAccount {@code true} if {@link Account} info has to be injected into the headers.
+   * @param insertContainer {@code true} if {@link Container} info has to be injected into the headers.
    * @throws JSONException
    */
-  private void setAmbryHeadersForPut(JSONObject headers, String ttlInSecs, String isPrivate, String serviceId,
-      String contentType, String ownerId) throws JSONException {
+  private void setAmbryHeadersForPut(JSONObject headers, String ttlInSecs, String serviceId, Container container,
+      String contentType, String ownerId, boolean insertAccount, boolean insertContainer) throws JSONException {
     headers.putOpt(RestUtils.Headers.TTL, ttlInSecs);
-    headers.putOpt(RestUtils.Headers.PRIVATE, isPrivate);
     headers.putOpt(RestUtils.Headers.SERVICE_ID, serviceId);
     headers.putOpt(RestUtils.Headers.AMBRY_CONTENT_TYPE, contentType);
     headers.putOpt(RestUtils.Headers.OWNER_ID, ownerId);
+    if (insertAccount) {
+      headers.putOpt(RestUtils.InternalKeys.TARGET_ACCOUNT_KEY, Account.UNKNOWN_ACCOUNT);
```

Comment:
It is a bit confusing that `insertAccount` controls inserting `UNKNOWN_ACCOUNT` and `insertContainer` controls inserting the given container.

Reply:
Same reply as before - the actual `Account` doesn't matter for this testing.